src/InvertedIndices.jl

```diff
@@ -8,8 +8,34 @@ struct InvertedIndex{S}
     skip::S
 end
 const Not = InvertedIndex
+
+nonBool2Int(x::Bool) = throw(ArgumentError("invalid index $x of type Bool"))
```

Comment:
Things like `Not(true)` and `Not(true, true)` are currently allowed, so shouldn't be keep allowing them?

Reply:
The fact that they are allowed is a bug in my opinion. What do you think `Not(true)` and `Not(true, ture)` should mean?

Reply:
The point is that for example I think that this is not what user wants:
```
julia> (1:2)[Not(true, true)]
1-element Vector{Int64}:
 2

julia> ([1,2])[Not(true, true)]
1-element Vector{Int64}:
 2
```

Reply:
I was thinking of non-array uses, with an object that would allow Boolean values as keys. `x[Not(true)]` would be a special case of `x[Not(inds)]` that should return `x[setdiff(keys(x), inds.skip)]`. But I have no idea whether such an object exists in a package. Do you think that makes sense?

Reply:
Just to be clear `Not(true)` is allowed. It will just error later.

What is not allowed in constructor is `Not(true, true)`.

See the tests:

```
@test_throws ArgumentError v[Not(true)]
@test_throws ArgumentError Not(1, true)
```

As you can notice `Not(true)` works. It only errors if you try to index a vector with it.

What fails is e.g. `Not(1, true)` (or similar). We disallow construction of such value as (this is before this PR):
```
julia> x = Not(true, true)
InvertedIndex{InvertedIndices.TupleVector{Tuple{Bool, Bool}}}([true, true])

julia> eltype(x.skip)
Int64
```
and as you can see these `true` values are treated as `Int`, which is a bug.

Is this clear now?